src/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixer.php

```diff
@@ -0,0 +1,264 @@
+        return 6;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_DOC_COMMENT);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $namespaceUseAnalyzer = new NamespaceUsesAnalyzer();
+
+        $shortNames = [];
+        foreach ($namespaceUseAnalyzer->getDeclarationsFromTokens($tokens) as $namespaceUseAnalysis) {
+            $shortNames[strtolower($namespaceUseAnalysis->getShortName())] = '\\'.strtolower($namespaceUseAnalysis->getFullName());
+        }
+
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind(T_DOC_COMMENT)) {
+                continue;
+            }
+
+            $functionIndex = $this->findDocumentedFunction($tokens, $index);
+            if (null === $functionIndex) {
+                continue;
+            }
+
+            $docBlock = new DocBlock($token->getContent());
+
+            $openingParenthesisIndex = $tokens->getNextTokenOfKind($functionIndex, ['(']);
+            $closingParenthesisIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $openingParenthesisIndex);
+
+            $argumentsInfo = $this->getArgumentsInfo(
+                $tokens,
+                $openingParenthesisIndex + 1,
+                $closingParenthesisIndex - 1
+            );
+
+            foreach ($docBlock->getAnnotationsOfType('param') as $annotation) {
+                if (0 === Preg::match('/@param\s+(?:\S|\s(?!\$))+\s(\$\S+)/', $annotation->getContent(), $matches)) {
+                    continue;
+                }
+
+                $argumentName = $matches[1];
+
+                if (
+                    !isset($argumentsInfo[$argumentName])
+                    || $this->annotationIsSuperfluous($annotation, $argumentsInfo[$argumentName], $shortNames)
+                ) {
+                    $annotation->remove();
+                }
+            }
+
+            $returnTypeInfo = $this->getReturnTypeInfo($tokens, $closingParenthesisIndex);
+
+            foreach ($docBlock->getAnnotationsOfType('return') as $annotation) {
+                if ($this->annotationIsSuperfluous($annotation, $returnTypeInfo, $shortNames)) {
+                    $annotation->remove();
+                }
+            }
+
+            $tokens[$index] = new Token([T_DOC_COMMENT, $docBlock->getContent()]);
+        }
+    }
+
+    private function findDocumentedFunction(Tokens $tokens, $index)
+    {
+        do {
+            $index = $tokens->getNextMeaningfulToken($index);
+
+            if ($tokens[$index]->isGivenKind(T_FUNCTION)) {
+                return $index;
+            }
+        } while ($tokens[$index]->isGivenKind([T_ABSTRACT, T_FINAL, T_STATIC, T_PRIVATE, T_PROTECTED, T_PUBLIC]));
+
+        return null;
+    }
+
+    private function getArgumentsInfo(Tokens $tokens, $start, $end)
```

Comment:
`ArgumentsAnalyzer` won't help here ?

Reply:
Didn't thought about that class. At first glance I'm not sure it can provide all information the fixer needs, but maybe it's worth adding a new method to it :)

Reply:
IIRC, for now it finds indexes of args. so not doing everything you need here, but half done